src/commands/environments/update.ts

```diff
@@ -10,33 +11,68 @@ import {
     EnvironmentPromptResult
 } from '../../ui/prompts'
 import UpdateCommand from '../updateCommand'
+import { Flags } from '@oclif/core'
 
-export default class UpdateEnvironment extends UpdateCommand<CreateEnvironmentParams> {
+export default class UpdateEnvironment extends UpdateCommand {
     static hidden = false
     static description = 'Update a Environment.'
 
     prompts = [
         namePrompt,
         descriptionPrompt
     ]
+    static flags = {
+        ...UpdateCommand.flags,
+        'type': Flags.string({
+            description: 'The type of environment',
+            options: environmentTypes
+        }),
+        'description': Flags.string({
+            description: 'Description for the environment',
+        }),
+    }
 
     public async run(): Promise<void> {
+        const { flags } = await this.parse(UpdateEnvironment)
+        const { key, name, type, description, headless } = flags
+
         await this.requireProject()
-        const { environment } = await inquirer.prompt<EnvironmentPromptResult>([environmentPrompt], {
-            token: this.authToken,
-            projectKey: this.projectKey
-        })
+        let envKey: string
+        if (headless) {
+            if (!key) {
```

Comment:
To be consistent with the new commands we're adding, the key or ID to identify the environment to update should be passed as an argument.
For many resources a user could pass both a key to identify the resource while also modifying the key. To differentiate between the two, we're setting the pattern that args are used to identify the resource and flags are used to pass additional values (in this case the properties to update)